From ca5adcdbd279d05ec61c3245d363678d05ee5714 Mon Sep 17 00:00:00 2001
From: Ben Widawsky <ben@bwidawsk.net>
Date: Wed, 11 May 2011 15:10:58 -0700
Patch-mainline: 3.0-rc3
References: fate#310916
Git-commit: 8eb572942ca02890f590d9251233038e27dd3842
Subject: [PATCH 2420/2588] drm/i915: forcewake debugfs fix

Forcewake needs to register itself with drm to use the remove function.

The file also should be read only.

Signed-off-by: Ben Widawsky <ben@bwidawsk.net>
Signed-off-by: Keith Packard <keithp@keithp.com>
(cherry picked from commit 8eb572942ca02890f590d9251233038e27dd3842)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_debugfs.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_debugfs.c b/drivers/gpu/drm/i915/i915_debugfs.c
index e753930..13d977b 100644
--- a/drivers/gpu/drm/i915/i915_debugfs.c
+++ b/drivers/gpu/drm/i915/i915_debugfs.c
@@ -1382,13 +1382,13 @@ static int i915_forcewake_create(struct dentry *root, struct drm_minor *minor)
 	struct dentry *ent;
 
 	ent = debugfs_create_file("i915_forcewake_user",
-				  S_IRWXU,
+				  S_IRUSR,
 				  root, dev,
 				  &i915_forcewake_fops);
 	if (IS_ERR(ent))
 		return PTR_ERR(ent);
 
-	return 0;
+	return drm_add_fake_info_node(minor, ent, &i915_forcewake_fops);
 }
 
 static struct drm_info_list i915_debugfs_list[] = {
-- 
1.7.6

